app/models/driver.rb

```diff
@@ -2,8 +2,9 @@ class Driver < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
 
-  devise :database_authenticatable, :rememberable, :trackable
-  validates :number, presence: true, uniqueness: true
-  validates_format_of :number, with: /\A(0)([0-9]{9})\z/
+  devise :database_authenticatable, :rememberable, :trackable, :authentication_keys => [:phone]
```

Comment:
Please use the Ruby 1.9 hash literal syntax when your hash keys are symbols. 
https://github.com/bbatsov/ruby-style-guide#hash-literals